flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/WatermarkSpec.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api;
+
+import org.apache.flink.table.types.DataType;
+
+import java.util.Objects;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Watermark metadata defined in {@link TableSchema}. It mainly contains two parts,
+ * 1) the rowtime attribute.
+ * 2) the watermark generation strategy.
+ */
+public class WatermarkSpec {
+
+	private final String rowtimeAttribute;
+
+	private final String watermarkExpressionString;
+
+	private final DataType watermarkExprOutputType;
+
+	public WatermarkSpec(String rowtimeAttribute, String watermarkExpressionString, DataType watermarkExprOutputType) {
+		this.rowtimeAttribute = checkNotNull(rowtimeAttribute);
+		this.watermarkExpressionString = checkNotNull(watermarkExpressionString);
+		this.watermarkExprOutputType = checkNotNull(watermarkExprOutputType);
+	}
+
+	/**
+	 * Returns the name of rowtime attribute, it can be a nested field using dot separator.
+	 * The referenced attribute must be present in the {@link TableSchema} and of
+	 * type {@link org.apache.flink.table.types.logical.LogicalTypeRoot#TIMESTAMP_WITHOUT_TIME_ZONE}.
+	 */
+	public String getRowtimeAttribute() {
+		return rowtimeAttribute;
+	}
+
+	/**
+	 * Returns the string representation of watermark generation expression.
+	 * The string representation is a qualified SQL expression string (UDFs are expanded).
+	 */
+	public String getWatermarkExpressionString() {
+		return watermarkExpressionString;
```

Comment:
`getWatermarkExpression` is okey i think.

Reply:
A `String` in the method name explicitly declares that the return value is a string representation instead of `Expression`.